gossip/filters/api.go

```diff
@@ -83,19 +84,25 @@ func NewPublicFilterAPI(backend Backend, cfg Config) *PublicFilterAPI {
 		events:  NewEventSystem(backend),
 		filters: make(map[rpc.ID]*filter),
 	}
-	go api.timeoutLoop(5 * time.Minute)
+	go timeoutLoop(weak.Make(api), 5*time.Minute)
```

Comment:
For the PR comment: keeping only a weak reference in the timeout-Loop is the actual reason why the API instance (and the Backend including the Carmen DB) can actually be finalized. In the old version, the timeout look kept those in use.